### ISSUE TITLE
[WebView] WebView does not persist session or authentication headers after intial page load

### ISSUE THREAD

**Issue by connor-odoherty** [expo/expo#6002]
### Environment

  Expo CLI 3.1.2 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.10.0 - /usr/local/bin/node
      Yarn: 1.17.3 - /usr/local/bin/yarn
      npm: 6.11.3 - /usr/local/bin/npm
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 11.0/11A420a - /usr/bin/xcodebuild
    npmPackages:
      @types/react: ^16.8.14 => 16.8.23 
      expo: ^35.0.0 => 35.0.0 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.1 => 3.11.1 
    npmGlobalPackages:
      expo-cli: 3.1.2

Target: iOS and Expo Client on iOS device and simulator

### Steps to Reproduce

1. Use a JWT token or session cookie to manage authentication state in a server side WebView
2. Set headers in the `source` prop
3. Start up the WebView and see in your server logs that the authentication headers and/or session cookies have been persisted. The first page should successfully authenticate
4. Navigate to the next page from the WebView using a 'PATCH' request
5. Receive authentication error and see in your server logs the headers you set were not persisted

### Expected Behavior

In Expo 34, this identical authentication flow worked as expected. You could enter the WebView and navigate from page to page without issue

### Actual Behavior

An authentication error appears and we see that the session cookie and auth headers were not persisted

### Reproducible Demo

Not easy to reproduce without providing a server for others to hit. However, it seems to be a well documented complaint.

Downgrading to Expo 34 seems to work, but that comes with a whole host of other issues, like WebView being completely broken when launched from an Expo Client app on both the iOS simulator and physical iOS devices with version 13+.

Even if there is not an answer to this exact problem, could you tell me what WebView version was used in the Expo 34 version of React Native? That would go a long way towards solving the issue

**Comment by connor-odoherty** [comment]
@cruzach I am importing from react-native-webview now. After doing some digging, it looks like this is an iOS 13 issue: https://bugs.webkit.org/show_bug.cgi?id=200857

tl;dr Cross-origin cookies (and apparently cookies in general) have been pretty broken with the release of iOS 13. Luckily, it looks like 13.2 should provide some fixes.

Anyway, thank you for letting me know which version 34 for was using! I'm considering downgrading because the react-native-webview 7 does not allow WebViews to load in the background.

Curious, how bad of an idea would it be (if it is even possible) to use an older version of react-native-webview with Expo 35? I know the packages are intentionally tightly coupled

**Comment by cruzach** [comment]
The one thing I would be aware of is if the version of `react-native-webview` you install uses `UIWebView`
> ITMS-90809: Deprecated API Usage - Apple will stop accepting submissions of apps that use UIWebView APIs . See https://developer.apple.com/documentation/uikit/uiwebview for more information.

**Comment by karimmorta** [comment]
Hi @connor-odoherty, Have you been able to make cookies work in your WebViews ? I'm using `react-native-webview 9.4.0` on an iOS 13.7 device but cookies are not persisted, after the first page it looks like.

**Comment by anasyo10** [comment]
Was this solved in any way :| having the same issue...

**Comment by Elue-dev** [comment]
any update on this please?

**Comment by DSp4wN** [comment]
I am also facing the similar issue for the first install if I close the my expo react native webview app in android within 30 seconds and start it will log me out (cookies are lost), but if I wait for more than 30 seconds after logging in then it will work properly (my assumption is that the cookies are not being flushed frequently)
